File: GATE ONE/DAYSEVEN/JAVA/DIARY/Diaries.java
import java.util.ArrayList;
import java.util.List;

public class Diaries {
    private final List<Diary> diaries = new ArrayList<>();

    public void add(String username, String password) {
        diaries.add(new Diary(username, password));
    }
    public Diary findByUsername(String username) {
        return diaries.stream().filter(diary -> diary.getUsername().equals(username)).findFirst().orElse(null);
    }
    public void delete(String username, String password) {
        Diary diary;
        diary = findByUsername(username);
        diary.unlockDiary(password);
        diaries.remove(diary);
    }
}

File: GATE ONE/DAYSEVEN/PYTHON/DIARY/Diaries.py
from Diary import Diary


class Diaries:
    def __init__(self):
        self.diaries = []

    def add(self, username, password):
        self.diaries.append(Diary(username, password))

    def findByUsername(self, username):
        for diary in self.diaries:
            if diary.getUsername() == username:
                return diary
        return None

    def delete(self, username, password):
        diary = self.findByUsername(username)
        diary.unlockDiary(password)
        self.diaries.remove(diary)